### PR TITLE
Remove the order by while exporting (fix #1372)

### LINKED ISSUE #1372: What's the point of the ORDER BY in export_public.php?
See [here](https://github.com/thewca/worldcubeassociation.org/blob/bf3edce49b52d0a01362f63d661142deb4dd256a/webroot/results/admin/export_public.php#L23-L24). I don't get what the point of this ORDER BY is, people are going to be inserting these rows into their own database anyways. This ORDER BY is very expensive for our export, so it would be nice to get rid of it.

@Baiqiang, what do you think?

### PR DESCRIPTION
**Pull request by Baiqiang**

### CODE REVIEW

**Comment by Baiqiang** [comment]
@FatBoyXPC Please give comments.

File: webroot/results/admin/export_public.php [hunk 1]
```diff
@@ -157,6 +154,10 @@ function exportPublic ( $sources ) {
         echo '.';  # shows both Apache and the user that the script is doing stuff and not hanging
       }
     }
+    //Check if any errors while fetching data
+    if (($error = mysql_error()) !== '') {
```
Comment (jfly) [inline webroot/results/admin/export_public.php]: Why stash the result of this function call in a `$error` variable if you don't go on to use it?

File: webroot/results/admin/export_public.php [hunk 2]
```diff
@@ -15,13 +15,10 @@
...
                                 best, average,
-                                personName, personId, result.countryId AS personCountryId,
+                                personName, personId, countryId AS personCountryId,
                                 formatId, value1, value2, value3, value4, value5,
                                 regionalSingleRecord, regionalAverageRecord
-                       FROM     Results result, Competitions competition, Events event, Rounds round
-                       WHERE    competition.id=competitionId AND event.id=eventId AND round.id=roundId
-                       ORDER BY competition.year, competition.month, competition.day, competition.id,
-                                event.rank, round.rank, pos, average, best, personName',
+                       FROM     Results',
```
Comment (jfly) [inline webroot/results/admin/export_public.php]: Woah! We can get rid of this crazy `join`-ing as well? Looks safe to me... why were we even doing this in the first place?

**Comment by jfly** [comment]
Could you make sure to test that the Workbook Assistant still works after this change?

File: webroot/results/admin/export_public.php (hunk 1 above)
Comment (jfly) [inline webroot/results/admin/export_public.php]: This script is only run by admins, so I think it's okay to dump sensitive information here. I'd be more worried about getting emailing working properly from php, lol.

Comment (FatBoyXPC) [inline webroot/results/admin/export_public.php]: Fair enough, I can get over that. I cannot, however, get over the idea that you're more concerned with getting email working. That's depressing :(

File: webroot/results/admin/export_public.php (hunk 2 above)
Comment (Baiqiang) [inline webroot/results/admin/export_public.php]: The purpose is to `order by`. We can remove it with `order by`.

File: webroot/results/admin/export_public.php (hunk 1 above)
Comment (Baiqiang) [inline webroot/results/admin/export_public.php]: I forgot to press a `Ctrl + S` and commit it. Forgive me. Changed it just now.

Comment (Baiqiang) [inline webroot/results/admin/export_public.php]: @FatBoyXPC I don't like the short condition. For now it looks more clear.

**Comment by Baiqiang** [comment]
@jfly The WA doesn't use the Results table, so it won't ben affected.

File: webroot/results/admin/export_public.php (hunk 1 above)
Comment (FatBoyXPC) [inline webroot/results/admin/export_public.php]: So what happens if `mysql_error()` returns boolean false? The docs say that this returns an empty string, but my PsySH REPL disagrees. Either way, I feel like just checking empty string could introduce a bug unless my REPL is just being weird.

Docs:
> Returns the error text from the last MySQL function, or '' (empty string) if no error occurred.

REPL:
```
>>> mysql_error()
=> false
>>> mysql_error() === ''
=> false
>>> mysql_error() !== ''
=> true
>>> 
```

Comment (Baiqiang) [inline webroot/results/admin/export_public.php]: @FatBoyXPC You SHOULD not run `mysql_error` without any mysql connection. The only case if return false is there's no mysql connection.

Comment (Baiqiang) [inline webroot/results/admin/export_public.php]: So it won't return false in our cases.

Comment (FatBoyXPC) [inline webroot/results/admin/export_public.php]: Gotcha! LGTM! :shipit:!

**Comment by FatBoyXPC** [comment]
:shipit: !

**Pull request merged**